x/poolmanager/router_test.go

```diff
@@ -1544,6 +1546,454 @@ func (s *KeeperTestSuite) TestSingleSwapExactAmountIn() {
+				PoolId:         poolId,
+				MaxPriceImpact: maxPriceImpact,
+				TwapPrice:      twapPriceOneConcentratedInv,
+			},
+			setPositionForCLPool: true,
+			setClTokens:          clCoinsLiquid,
+			expectedInputCoin:    sdk.NewCoin(assetUsdc, sdk.NewInt(1)),
+			expectedOutputCoin:   sdk.NewCoin(assetEth, sdk.NewInt(0)),
+		},
+		"valid concentrated pool - twap value much greater than spot price do not trade": {
+			preCreatePoolType: types.Concentrated,
+			poolId:            poolId,
+			req: queryproto.EstimateTradeBasedOnPriceImpactRequest{
+				FromCoin:       sdk.NewCoin(assetEth, sdk.NewInt(30_000)),
+				ToCoinDenom:    assetUsdc,
+				PoolId:         poolId,
+				MaxPriceImpact: maxPriceImpact,
+				TwapPrice:      twapPriceTwoConcentrated,
+			},
+			setPositionForCLPool: true,
+			setClTokens:          clCoinsLiquid,
+			expectedInputCoin:    sdk.NewCoin(assetEth, sdk.NewInt(0)),
+			expectedOutputCoin:   sdk.NewCoin(assetUsdc, sdk.NewInt(0)),
+		},
+		"valid concentrated pool - adjusted price impact halved": {
+			preCreatePoolType: types.Concentrated,
+			poolId:            poolId,
+			req: queryproto.EstimateTradeBasedOnPriceImpactRequest{
+				FromCoin:       sdk.NewCoin(assetEth, sdk.NewInt(30_000)),
+				ToCoinDenom:    assetUsdc,
+				PoolId:         poolId,
+				MaxPriceImpact: maxPriceImpactTiny,
+				TwapPrice:      twapPriceOneConcentrated,
+			},
+			setPositionForCLPool: true,
+			setClTokens:          clCoinsLiquid,
+			expectedInputCoin:    sdk.NewCoin(assetEth, sdk.NewInt(10_733)),
+			expectedOutputCoin:   sdk.NewCoin(assetUsdc, sdk.NewInt(53_638_181)),
+		},
+		"valid concentrated pool - twap price halfs adjusted price impact": {
+			preCreatePoolType: types.Concentrated,
+			poolId:            poolId,
+			req: queryproto.EstimateTradeBasedOnPriceImpactRequest{
+				FromCoin:       sdk.NewCoin(assetEth, sdk.NewInt(30_000)),
+				ToCoinDenom:    assetUsdc,
+				PoolId:         poolId,
+				MaxPriceImpact: maxPriceImpact,
+				TwapPrice:      twapPriceThreeConcentrated,
+			},
+			setPositionForCLPool: true,
+			setClTokens:          clCoinsLiquid,
+			expectedInputCoin:    sdk.NewCoin(assetEth, sdk.NewInt(10_746)),
+			expectedOutputCoin:   sdk.NewCoin(assetUsdc, sdk.NewInt(53_703_116)),
+		},
+		"valid concentrated pool - adjusted price impact halved - twap not given": {
+			preCreatePoolType: types.Concentrated,
+			poolId:            poolId,
+			req: queryproto.EstimateTradeBasedOnPriceImpactRequest{
+				FromCoin:       sdk.NewCoin(assetEth, sdk.NewInt(30_000)),
+				ToCoinDenom:    assetUsdc,
+				PoolId:         poolId,
+				MaxPriceImpact: maxPriceImpactTiny,
+				TwapPrice:      sdk.ZeroDec(),
+			},
+			setPositionForCLPool: true,
+			setClTokens:          clCoinsLiquid,
+			expectedInputCoin:    sdk.NewCoin(assetEth, sdk.NewInt(10_733)),
+			expectedOutputCoin:   sdk.NewCoin(assetUsdc, sdk.NewInt(53_638_181)),
+		},
+		"valid concentrated pool - liquidity too low token out estimation is 0": {
```

Comment:
I think this is an invalid test case - we end up getting a zero because `if adjustedMaxPriceImpact.IsZero() || adjustedMaxPriceImpact.IsNegative()` is satisfied.

Also, don't we expect an error if there's not enough liquidity?

Perhaps the name of the test case is what's wrong here.